### ISSUE TITLE
Secrets file permissions changing from 0400 to 0440

### ISSUE THREAD

**Issue by mlangbehn** [BonnyCI/projman#154]
During an [ansible run][1], we can see the POSIX permissions on `"{{ secret_file_path }}"` changing from `root:root 0400` to `root:root 0440`. [Searching the hoist repository for '0440'][2] returns no results, but we are modifying the permissions three times during the [ansible run][1].

The [first place we modify them][3] uses the [ansible file module][4] to set the POSIX permissions to `root:root 0400`, the [second place we modify them][5] uses the [ansible ACL module][6] to set a base ACL (setting the mask to read-only), and the [third place we modify them][7] also uses the [ansible ACL module][6] to add a read-only ACE for the `"{ secret_file_user }"` (which gets set to the `"{ ansible_runner_user }"` [here][8]).

When `TASK [ansible-runner : check secrets exist]` is run during the [ansible run][1] (search the [log][1] for '0440'), we can see that the secrets file's permissions are being set to `root:root 0440` after being reset to `root:root 0400`.

Since the group on this file is `root`, I do not see any pressing security concerns, but it is rather annoying to have the permissions changing when they should not be. This is also causing us to have two roles reporting as `changed` twice during each deployment (see BonnyCI/projman#38 for the main issue tracking this).

[1]: http://contrasjc-bastion.portbleu.com/cron-logs/system-ansible/ansible_bastion.yml_20170303154501.log
[2]: https://github.com/BonnyCI/hoist/search?utf8=%E2%9C%93&q=0440
[3]: https://github.com/BonnyCI/hoist/blob/master/roles/ansible-runner/tasks/secret-file.yml#L6
[4]: https://docs.ansible.com/ansible/file_module.html
[5]: https://github.com/BonnyCI/hoist/blob/master/roles/ansible-runner/tasks/secret-file.yml#L17
[6]: https://docs.ansible.com/ansible/acl_module.html
[7]: https://github.com/BonnyCI/hoist/blob/master/roles/ansible-runner/tasks/secret-file.yml#L28
[8]: https://github.com/BonnyCI/hoist/blob/master/roles/ansible-runner/tasks/main.yml#L95

**Comment by missaugustina** [comment]
Is this issue resolved with that last PR that was merged or is there still additional work to do on this issue? If so, what?

**Comment by mlangbehn** [comment]
This is still in progress. It was on hold while I was working on getting vagrant/docker deploys working to continue testing.

**Comment by mlangbehn** [comment]
~Blocked by:~
* ~BonnyCI/hoist#242~
* ~BonnyCI/hoist#243~

**Comment by mlangbehn** [comment]
Confirmed that we are expecting the correct behavior by testing manually:

```text
ubuntu@ubuntu-xenial:~$ sudo useradd foo
ubuntu@ubuntu-xenial:~$ sudo touch /bar
ubuntu@ubuntu-xenial:~$ ls -al /bar
-rw-r--r-- 1 root root 0 Apr  7 14:19 /bar
ubuntu@ubuntu-xenial:~$ sudo chown ubuntu:ubuntu /bar
ubuntu@ubuntu-xenial:~$ chmod 0400 /bar
ubuntu@ubuntu-xenial:~$ ls -al /bar
-r-------- 1 ubuntu ubuntu 0 Apr  7 14:19 /bar
ubuntu@ubuntu-xenial:~$ getfacl /bar
getfacl: Removing leading '/' from absolute path names
# file: bar
# owner: ubuntu
# group: ubuntu
user::r--
group::---
other::---

ubuntu@ubuntu-xenial:~$ setfacl -m m:r /bar
ubuntu@ubuntu-xenial:~$ ls -l /bar
-r--r-----+ 1 ubuntu ubuntu 0 Apr  7 14:19 /bar
ubuntu@ubuntu-xenial:~$ getfacl /bar
getfacl: Removing leading '/' from absolute path names
# file: bar
# owner: ubuntu
# group: ubuntu
user::r--
group::---
mask::r--
other::---

ubuntu@ubuntu-xenial:~$ setfacl -m u:foo:r /bar
ubuntu@ubuntu-xenial:~$ ls -l /bar
-r--r-----+ 1 ubuntu ubuntu 0 Apr  7 14:19 /bar
ubuntu@ubuntu-xenial:~$ getfacl /bar
getfacl: Removing leading '/' from absolute path names
# file: bar
# owner: ubuntu
# group: ubuntu
user::r--
user:foo:r--
group::---
mask::r--
other::---
```

**Comment by mlangbehn** [comment]
While @j2sol and I took a look at this issue together, we found that issue is with `setfacl` modifying the POSIX permissions.

According to [RedHat's documentation on POSIX ACLs][1], this is expected behavior. As such, I am now testing a change to have us treat permissions the way our systems are treating them.

[1]:https://access.redhat.com/documentation/en-US/Red_Hat_Storage/2.0/html/Administration_Guide/ch09s05.html

**Comment by mlangbehn** [comment]
Issue resolved.
